### DISCUSSION TITLE
Possible to make expandable left sidebar items not correspond to pages?

### DISCUSSION THREAD

**Discussion by asmeurer** [pradyunsg/furo#538] [Questions]
In the SymPy docs, we have a lot of expandable sections in the left sidebar that effectively only serve the purpose of splitting the subpages into useful categories. The index pages themselves are not particularly useful, at least in their current state, as they typically only contain a toctree. For example https://docs.sympy.org/latest/reference/public/basics/index.html

Now no doubt we could improve things by adding some text to those index pages, but I'm wondering if it's possible to somehow make it so that these sections don't correspond to pages at all. Rather, clicking an expandable section name in the left sidebar would just expand the section. 

This is how the GitHub docs, which have a similar layout to Furo, work. For instance, click on the expandable items at https://docs.github.com/en/actions/using-workflows/about-workflows. They don't link to a specific page, but rather just expand or collapse the sections. 

I also feel like this would improve the general usability of the left sidebar. I often find myself navigating the left sidebar by clicking on the index page titles, even though I know this will incur a page load each time, because they are larger targets and easier to click on than the little arrows. 

Is it possible to do this? Presumably one could hack this by changing the theme to make the sidebar titles do the same thing as the arrows, but the toctree index pages would still exist. Can they be removed entirely, or hidden in Sphinx somehow?

**Comment by pradyunsg** [comment]
This isn't technically possible with Sphinx today, but I'm open to ideas for this!

Sphinx's toctree data structure and markup only supports representations in a manner incompatible with the needs expressed here. If you can figure out some way to have the markup contain not-a-link in the element, I'd be more than happy to tweak the generated markup from Furo to collapse on the whole thing. This might actually end up being a feature in Lutra instead, FWIW -- although I ain't promising timelines.

I've been wondering if it would be useful to Ajaxify the page loads, so that things feel snappier; but I've not decided yet whether it's a good idea or a bad one.

**Comment by asmeurer** [comment]
My understanding of Sphinx is that it assumes there is an exact one-to-one correspondence between documents and files, and this is pretty hard-coded. So I rather doubt it will be possible to completely remove the index document. You can set `:hidden:` on `.. toctree` but that only hides the table of contents itself, not the whole document. 

I think instead one would need to indicate with some sort of metadata on the page that it should be hidden in the sidebar. The page itself would still exist, but there would be no way within the theme navigation to actually access it. I don't know what the best way to approach that would be. Perhaps using the [meta directive](https://docutils.sourceforge.io/docs/ref/rst/directives.html#metadata)?

**Reply by pradyunsg** [reply]
Well, https://pradyunsg.me/furo/customisation/#page-specific-tweaks would be a smoother way to do things (I think that's what you were intending to refer to?).

The bigger problem is that we don't have that metadata available in the markup or anything. I guess one option is to rewrite https://github.com/pradyunsg/furo/blob/821267c402f369e3c115368a1cd531feb299d8f3/src/furo/navigation.py#L20 completely to do things differently.
